apps/cow-fi/components/SearchBar.tsx

```diff
@@ -213,25 +312,125 @@ export const SearchBar: React.FC<SearchBarProps> = ({ articles }) => {
   }, [isFocused, highlightedIndex, filteredArticles])
 
   useEffect(() => {
-    if (query.trim()) {
-      const filtered = (articles || []).filter((article) =>
-        article.attributes?.title?.toLowerCase().includes(query.toLowerCase()),
-      )
+    if (debouncedQuery.trim()) {
+      const searchTerm = debouncedQuery.toLowerCase()
+
+      // TODO: Performance consideration
+      // This multi-criteria search implementation works well for moderate datasets,
+      // but may become expensive for large article collections.
+      // If scaling is anticipated, consider implementing:
+      // 1. An indexing-based approach (e.g., Elasticlunr, Fuse.js)
+      // 2. Server-side search with pagination
+      const filtered = (articles || []).filter((article) => {
+        const title = article.attributes?.title?.toLowerCase() || ''
+        const description = article.attributes?.description?.toLowerCase() || ''
+        const slug = article.attributes?.slug?.toLowerCase() || ''
+
+        // Track if we've found a match to avoid redundant checks
+        let foundMatch = false
+
+        // 1. Check if the entire search term is contained in the title, description, or slug
+        if (title.includes(searchTerm) || description.includes(searchTerm) || slug.includes(searchTerm)) {
+          foundMatch = true
```

Comment:
It doesn't make sense to set `foundMatch = true` because in the next line we exit from the iteration

Reply:
Might be outdated comment now